plugin.py

```diff
@@ -0,0 +1,88 @@
+from LSP.plugin import AbstractPlugin, ClientConfig, register_plugin, unregister_plugin, WorkspaceFolder
+from LSP.plugin.core.typing import Dict, Optional, List
+from shutil import which
+import os
+import sublime
+import urllib.request
+
+
+MARKSMAN_TAG = '2022-08-27'
+MARKSMAN_RELEASES_BASE = 'https://github.com/artempyanykh/marksman/releases/download/{tag}/{platform}'
+USER_AGENT = 'Sublime Text LSP'
+
+
+def marksman_binary() -> Optional[str]:
+    platform = sublime.platform()
+    if platform == 'osx':
+        return 'marksman-macos'
+    if platform == 'windows':
+        return 'marksman.exe'
+    if platform == 'linux':
+        return 'marksman-linux'
+    return None
```

Comment:
```suggestion
def marksman_binary() -> Optional[str]:
    platform_arch = "{}_{}".format(sublime.platform(), sublime.arch())
    if platform_arch == 'osx_x64':
        return 'marksman-macos'
    if platform_arch == 'windows_x64':
        return 'marksman.exe'
    if platform_arch == 'linux_x64':
        return 'marksman-linux'
    return None
```

Those pre-builts are x64 https://github.com/artempyanykh/marksman/runs/8049673045

![image](https://user-images.githubusercontent.com/6594915/187059032-bc139ceb-7b2f-4fb9-8cdb-dc0e243c4e3a.png)
![image](https://user-images.githubusercontent.com/6594915/187058992-9a5f3ebe-b596-4bb0-ad96-321c50bc4b9d.png)
![image](https://user-images.githubusercontent.com/6594915/187059002-3ba35e66-3267-492a-a63d-6ffffa05c5e1.png)

Reply:
It still work on arm Mac through emulation. Better than nothing.

Reply:
suggestion revised.

Reply:
But makes sense to do for other platforms

Reply:
Suggestion revised.